Core/Epaper/Src/epaper.c

```diff
@@ -0,0 +1,462 @@
+	sendBuffer[0] = lut[155];
+	sendBuffer[1] = lut[156];
+	sendBuffer[2] = lut[157];
+	sendMessage(CMD_SOURCE_DRIVING_VOLTAGE_CONTROL, sendBuffer, 3);
+
+	// Set VCOM Register
+	sendBuffer[0] = lut[158];
+	sendMessage(CMD_WRITE_VCOM_REGISTER, sendBuffer, 1);
+}
+
+/*!
+  @brief   Turn on Epaper display
+*/
+static void turnOnDisplay()
+{
+	// Activate display update sequence
+	sendCommand(CMD_MASTER_ACTIVATION);
+
+	// Wait for busy_release interupt
+	waitBusyRelease();
+}
+
+/*!
+  @brief	Set Window Size of epaper
+  @param	Xstart	Starting X coordinate
+  @param	Ystart	Starting Y coordinate
+  @param	Xend 	Ending X coordinate
+  @param	Yend	Ending Y coordinate
+*/
+static void setWindows(uint16_t Xstart, uint16_t Ystart, uint16_t Xend, uint16_t Yend)
+{
+	// SET_RAM_X_ADDRESS_START_END_POSITION
+	sendBuffer[0] = (Xstart>>3) & 0xFF;
+	sendBuffer[1] = (Xend>>3) & 0xFF;
+	sendMessage(CMD_SET_RAM_X_WINDOW, sendBuffer, 2);
+	
+	// SET_RAM_Y_ADDRESS_START_END_POSITION
+	sendBuffer[0] = Ystart & 0xFF;
+	sendBuffer[1] = (Ystart >> 8) & 0xFF;
+	sendBuffer[2] = Yend & 0xFF;
+	sendBuffer[3] = (Yend >> 8) & 0xFF;
+    sendMessage(CMD_SET_RAM_Y_WINDOW, sendBuffer, 4);
+}
+
+/*!
+  @brief	Set Cursor position of epaper
+  @param	Xstart	Starting X coordinate
+  @param	Ystart	Starting Y coordinate
+*/
+static void setCursor(uint16_t Xstart, uint16_t Ystart)
+{
+	// SET_RAM_X_ADDRESS_COUNTER
+	sendBuffer[0] = (Xstart & 0xFF);
+	sendMessage(CMD_SET_RAM_X_COUNTER, sendBuffer, 1);
+
+	// SET_RAM_Y_ADDRESS_COUNTER
+	sendBuffer[0] = Ystart & 0xFF;
+	sendBuffer[1] = ((Ystart >> 8) & 0xFF);
+    sendMessage(CMD_SET_RAM_Y_COUNTER, sendBuffer, 2);
+}
+
+/*!
+  @brief	Initialize epaper display
+*/
+static void setFullRefreshSettings()
+{
+	// Set initial pin configuration
+	setCommandMode();
+	csOn();
+	enable();
+
+	// Excecute delayed reset sequence
+	vTaskDelay(100 / portTICK_PERIOD_MS);
+	reset();
+	enable();
+	vTaskDelay(100 / portTICK_PERIOD_MS);
+	waitBusyRelease();
+
+	// Software reset
+	sendCommand(CMD_SW_RESET); 
+	waitBusyRelease();   
+
+	// Configure driver output control
+	sendBuffer[0] = 0x27;
+	sendBuffer[1] = 0x01;
+	sendBuffer[2] = 0x00;
+	sendMessage(CMD_DRIVER_OUTPUT_CONTROL, sendBuffer, 3);   
+
+	// Configure Data entry mode
+	sendBuffer[0] = 0x03;
+	sendMessage(CMD_DATA_ENTRY_MODE_SETTING, sendBuffer, 1); 
+
+	// Set window border size
+	setWindows(0, 0, EPD_WIDTH-1, EPD_HEIGHT-1);
+
+	// Configure display update control register 1
+	sendBuffer[0] = 0x00;
+	sendBuffer[1] = 0x80;
+	sendMessage(CMD_DISPLAY_UPDATE_CONTROL_1, sendBuffer, 2);
+
+	// Set initial cursor position
+	setCursor(0, 0);
+
+	// Wait for busy_release interupt
+	waitBusyRelease();
+
+	// Set initial lookup table register 
+	setLookupTable(WS_20_30_LUT);
+
+    // Set display update sequence
+	sendBuffer[0] = 0xc7;
+	sendMessage(CMD_DISPLAY_UPDATE_CONTROL_2, sendBuffer, 1);
+
+	epdConfig = FULL_REFRESH;
+}
+
+static void setPartialRefreshSettings()
+{
+	// Reset epaper communication
```

Comment:
Same as above comment - do we want to add a check that we are not already in the correct setting?